res/controllers/Numark-Scratch-scripts.js

```diff
@@ -0,0 +1,452 @@
+    };
+
+    this.setMode = function(_channel, control) {
+        var newMode = this.modes[control];
+        this.currentMode.forEachComponent(function(component) {
+            component.disconnect();
+        });
+
+        newMode.forEachComponent(function(component) {
+            component.connect();
+            component.trigger();
+        });
+
+        this.currentMode = newMode;
+    };
+
+    this.currentMode = this.modes[NumarkScratch.PadModeControls.HOTCUE];
+};
+NumarkScratch.PadSection.prototype = Object.create(components.ComponentContainer.prototype);
+
+NumarkScratch.ModeHotcue = function(deckNumber) {
+    components.ComponentContainer.call(this);
+
+    this.control = NumarkScratch.PadModeControls.HOTCUE;
+
+    this.pads = new components.ComponentContainer();
+    for (var i = 0; i < 4; i++) {
+        this.pads[i] = new components.HotcueButton({
+            group: "[Channel" + deckNumber + "]",
+            midi: [0x93 + deckNumber, 0x14 + i],
+            number: i + 1,
+            shiftControl: true,
+            sendShifted: true,
+            shiftOffset: 0x08,
+            outConnect: false
+        });
+    }
+};
+NumarkScratch.ModeHotcue.prototype = Object.create(components.ComponentContainer.prototype);
+
+NumarkScratch.ModeRoll = function(deckNumber) {
+    components.ComponentContainer.call(this);
+
+    this.control = NumarkScratch.PadModeControls.ROLL;
+
+    this.pads = new components.ComponentContainer();
+    for (var i = 0; i < 4; i++) {
+        this.pads[i] = new components.Button({
+            group: "[Channel" + deckNumber + "]",
+            midi: [0x93 + deckNumber, 0x14 + i],
+            size: NumarkScratch.autoLoopSizes[i],
+            shiftControl: true,
+            sendShifted: true,
+            shiftOffset: 0x08,
+            inKey: "beatlooproll_" + NumarkScratch.autoLoopSizes[i] + "_activate",
+            outKey: "beatloop_" + NumarkScratch.autoLoopSizes[i] + "_enabled",
+            outConnect: false
+        });
+    }
+};
+NumarkScratch.ModeRoll.prototype = Object.create(components.ComponentContainer.prototype);
+
+NumarkScratch.ModeSampler = function(deckNumber) {
+    components.ComponentContainer.call(this);
+
+    this.control = NumarkScratch.PadModeControls.SAMPLER;
+
+    this.pads = new components.ComponentContainer();
+    for (var i = 0; i < 4; i++) {
+        this.pads[i] = new components.SamplerButton({
+            midi: [0x93 + deckNumber, 0x14 + i],
+            number: i + 1,
+            shiftControl: true,
+            sendShifted: true,
+            shiftOffset: 0x08,
+            outConnect: false
+        });
+    }
+};
+NumarkScratch.ModeSampler.prototype = Object.create(components.ComponentContainer.prototype);
+
+NumarkScratch.Gains = function() {
+    this.cueMix = new components.Pot({
+        group: "[Master]",
+        inKey: "headMix"
+    });
+};
+NumarkScratch.Gains.prototype = new components.ComponentContainer();
```

Comment:
this componentContainer is unnecessary. Just put the cuemix into the controller.

Reply:
Done thanks [24d9eb7](https://github.com/mixxxdj/mixxx/pull/4834/commits/24d9eb7a7d32e0743f634bbc9db9347c96b1154c)